base_layer/core/src/base_node/comms_interface/inbound_handlers.rs

```diff
@@ -252,6 +251,82 @@ where T: BlockchainBackend + 'static
                 }
                 Ok(NodeCommsResponse::HistoricalBlocks(blocks))
             },
+            NodeCommsRequest::FetchBlocksWithKernels(excess_sigs) => {
+                let mut blocks = Vec::<HistoricalBlock>::with_capacity(1);
```

Comment:
Did you perhaps mean `with_capacity(excess_sigs.length())`?

Reply:
Good catch, thats what it was, I played around with that code as I has other problems. Fixed this.